### PR TITLE
MNT-22233 resolved error while loading subcategory with cutoff.

### PR DESCRIPTION
**Pull request by brijmohan1**
Resolved error while loading records subcategory with cutoff.

### CODE REVIEW

**Review by kavitshah-gl — APPROVED** [review]
This is fine.

**Review by tpage-alfresco — APPROVED** [review]
This is a definite improvement.

Should we add some logging too?  What does it mean if `dispositionActionDefinition` is null?

**Comment by brijmohan1** [comment]
> This is a definite improvement.
> 
> Should we add some logging too? What does it mean if `dispositionActionDefinition` is null?

**Comment by brijmohan1** [comment]
added the log.

File: amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/jscript/app/JSONConversionComponent.java [hunk 1]
```diff
@@ -76,6 +79,7 @@
  *
  * @author Roy Wetherall
  */
+@Slf4j
```
Comment (tpage-alfresco) [inline amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/jscript/app/JSONConversionComponent.java:82]: This is cool - I've not seen this annotation before.

File: amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/jscript/app/JSONConversionComponent.java [hunk 2]
```diff
@@ -515,17 +519,25 @@ public Map<String, Object> doWork() throws Exception
...
                     {
+                        DispositionActionDefinition dispositionActionDefinition = nextDispositionAction.getDispositionActionDefinition();
                         HashMap properties = (HashMap) rmNodeValues.get("properties");
-                        properties.put("combineDispositionStepConditions", nodeService.getProperty(dispositionService.getNextDispositionAction(nodeRef).getDispositionActionDefinition().getNodeRef(), PROP_COMBINE_DISPOSITION_STEP_CONDITIONS));
                         properties.put("incompleteDispositionEvent", details.getEventName());
-                        properties.put("dispositionEventCombination", nodeService.getProperty(dispositionService.getNextDispositionAction(nodeRef).getDispositionActionDefinition().getNodeRef(), PROP_DISPOSITION_EVENT_COMBINATION));
-
+                        if(dispositionActionDefinition == null)
+                        {
+                            log.debug("Disposition action definition for this node has been removed or never exist");
```
[the remark this answers is not in the document]
Comment (brijmohan1) [reply amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/jscript/app/JSONConversionComponent.java]: done

File: amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/jscript/app/JSONConversionComponent.java (hunk 1 above)
Comment (brijmohan1) [reply amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/jscript/app/JSONConversionComponent.java:82]: Thank you for noticing.

**Pull request merged**